src/dsl/step.ts

```diff
@@ -23,28 +23,37 @@
  *
  */
 
-import { Location, VoidCallback } from '../common_types';
+import { Location, StatusValue, VoidCallback } from '../common_types';
 
 export class Step {
-  name: string;
-  index: number;
-  callback: VoidCallback;
-  location?: Location;
+  readonly name: string;
+  readonly index: number;
+  readonly location?: Location;
+  #cb: VoidCallback;
   skip = false;
   soft = false;
   only = false;
+  _startTime = 0;
+  duration = -1;
+  status: StatusValue = 'succeeded';
```

Comment:
nit: I feel something in the lines of `pending` would be a more sensible default for this.

Reply:
should be fine for UI

Reply:
Done https://github.com/elastic/synthetics/pull/960/commits/479970ce2e2bfb70e1926de6e58c5728a7d6cce8